cmd/file/io.go

```diff
@@ -0,0 +1,235 @@
+// Copyright 2020 The Swarm Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style
+// license that can be found in the LICENSE file.
+
+package file
+
+import (
+	"bytes"
+	"context"
+	"errors"
+	"fmt"
+	"io"
+	"io/ioutil"
+	"net/http"
+	"net/url"
+	"path/filepath"
+	"strings"
+
+	"github.com/sirupsen/logrus"
+	"github.com/spf13/cobra"
+
+	"github.com/ethersphere/bee/pkg/file"
+	"github.com/ethersphere/bee/pkg/logging"
+	"github.com/ethersphere/bee/pkg/storage"
+	"github.com/ethersphere/bee/pkg/swarm"
+)
+
+// nopWriteCloser wraps a io.Writer in the same manner as ioutil.NopCloser does
+// with an io.Reader.
+type nopWriteCloser struct {
+	io.Writer
+}
+
+// NopWriteCloser returns a new io.WriteCloser with the given writer as io.Writer.
+func NopWriteCloser(w io.Writer) io.WriteCloser {
+	return &nopWriteCloser{
+		Writer: w,
+	}
+}
+
+// Close implements io.Closer
+func (n *nopWriteCloser) Close() error {
+	return nil
+}
+
+// putGetter wraps both storage.Putter and storage.Getter interfaces
+type putGetter interface {
+	storage.Putter
+	storage.Getter
+}
+
+// TeeStore provides a storage.Putter that can put to multiple underlying storage.Putters.
+type TeeStore struct {
+	putters []storage.Putter
+}
+
+// NewTeeStore creates a new TeeStore.
+func NewTeeStore() *TeeStore {
+	return &TeeStore{}
+}
+
+// Add adds a storage.Putter.
+func (t *TeeStore) Add(putter storage.Putter) {
```

Comment:
why not in the constructor?
```
NewTeeStore(putters ...storage.Putter) *TeeStore {

}
```

Also, `NewTeeStore` should probably return a `storage.Putter`. No need to return the concrete type here. then `TeeStore` type can be unexported

Reply:
I find this easier to read

Reply:
> Also, NewTeeStore should probably return a storage.Putter.

:+1:

Reply:
Actually, if you look at `cmd/bee-split/main.go` you can see why having `Add` would be the better interface here.

Reply:
well you could have alternatively:
```go
var stores []storage.Storer
if file { stores = append(stores, NewFileStore()) }
if http { stores = append(stores, NewHttpStore()) }
tee := NewTeeStore(stores...)
```
but this is a terribly tyrannic and selective-vision nitpick :)

Reply:
Aswell as ugly as f*****ck ;)